tests/unit/bitops.tcl

```diff
@@ -82,21 +95,49 @@ start_server {tags {"bitops"}} {
             }
             assert {[r bitcount str $start $end] == [count_bits [string range $str $start $end]]}
         }
+
+        for {set j 0} {$j < 100} {incr j} {
+            set str [randstring 0 3000]
+            r set str $str
+            set l [expr [string length $str] * 8]
+            set start [randomInt $l]
+            set end [randomInt $l]
+            if {$start > $end} {
+                lassign [list $end $start] start end
+            }
+            assert {[r bitcount str $start $end bit] == [count_bits_start_end $str $start $end]}
```

Comment:
i looked at your fuzz tester, and IIUC it covers that case.
right?

i.e. instead of just 80*4 tests it runs some 2000, and but it actually has a chance to maybe achieve more than what i aimed for.
on the other hand, since it's a fuzzer, it may sometimes succeed and other time fail.
maybe if we reduce the string length there from 64k to 30, it can achieve more, and will actually be faster (BITPOS is an O(n) operation).

Reply:
Yes. But just `BITPOS`.

Reply:
Why should I use `B*`, for byte index, the bit order is not the matter. But for bit index, `b*` leads to error.

Reply:
IIRC the difference between `B` and `b` is not the byte order, but the bit order. in one the bits are `high-to-low order within each byte` and in the other `low-to-high`.
of course changing to `b` will fail your test, unless you also change the order in the command's C code.
but that's exactly my point... it's good that we test our code against a neutral software that's not affected by our bugs, but i also want to test BITPOS and BITCOUNT order against SETBIT, so we won't have an inconsistency about the order of bits in the byte between the various commands.
i.e. i imagine that the `216` value in the following line is something you got empirically (not on a whiteboard), so it can serve for spotting a regression, but not to see that the code is correct.
```tcl
assert {[r bitpos str 1 17 -1 bit] == 216}
```

let's improve your fuzzser to include BITCOUNT in addition to BITPOS, and reduce the size of the string (in theory 3 bytes are enough, but let's take at least 10 (since BITPOS implementation has some code that processes full words differently)